FrameworkSystem/DB/InstalledComponentsDB.py

```diff
@@ -0,0 +1,702 @@
+"""
+Classes and functions for easier management of the InstalledComponents database
+"""
+
+__RCSID__ = "$Id$"
+
+import datetime
+from sqlalchemy import *
+from sqlalchemy.orm import session, sessionmaker, mapper, relationship
+from sqlalchemy.orm.collections import InstrumentedList
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.sql import *
+from sqlalchemy.engine.reflection import Inspector
+from DIRAC import gConfig, gLogger, S_OK, S_ERROR
+from DIRAC.ConfigurationSystem.Client.PathFinder import getDatabaseSection
+
+metadata = MetaData()
+Base = declarative_base()
+
+class Component( Base ):
+  __tablename__ = 'Components'
+  __table_args__ = {
+    'mysql_engine': 'InnoDB',
+    'mysql_charset': 'utf8'
+  }
+
+  ComponentID = Column( 'ComponentID', Integer, primary_key = True )
+  System = Column( 'System', String( 32 ), nullable = False )
+  Module = Column( 'Module', String( 32 ), nullable = False )
+  Type = Column( 'Type', String( 32 ), nullable = False )
+
+  def __init__( self, system = null(), module = null(), cType = null() ):
+    self.System = system
+    self.Module = module
+    self.Type = cType
+
```

Comment:
We usually have camelcase convention (starting with lowercase) for variables. So these would be self.system instead of self.System, and so on for the others. 
Another minor comment: self.type is a bit too generic, just call it self.cType?